### ISSUE TITLE
SubjectsDataset integration for GridSampler?

### ISSUE THREAD

**Issue by potpov** [fepegar/torchio#538]
hello, thanks for your great work!
it looks to me that there's not a way to load an entire set of Subjects into a single Gridsampler entity. 
so far I created multiple Gridsamplers for each patient in my test-set. am I missing something? are you going to update Gridsampler to wrap multiple Subjects? thanks.

**Comment by romainVala** [comment]
what the problem with a gridsampler for each subjects ?

**Comment by potpov** [comment]
A discussion was probably going to better fit this topic, sorry about that. 
I saw in #467  someone already faced this matter.
I think there is nothing really wrong with this approach, it works. 
it just looks to me theoretically more elegant if we could wrap all subjects in a single torch dataloader object instead of having hundreds of them when your dataset is large. 
would be nice to see that in a future release.

**Comment by fepegar** [comment]
Hey, @potpov. Can you tell us a bit more about your use case? Is this to be used with the `Queue`?

**Comment by potpov** [comment]
> 
> 
> Hey, @potpov. Can you tell us a bit more about your use case? Is this to be used with the `Queue`?

Thank you for your reply and interest in my set-up. 
I'm using the GridSampler both with and without queue. 
my dataset is made out of 3D volumes with very different shapes.
I reshaped both data and gt volumes for my training set while I left GT volumes at their original shapes in test set. 

training: I load my data from a queue which I fed with a generic GridSampler: then I compare subvolumes directly for computing metrics and losses.
test: I created one GridSampler and DataLoader per patient in the test set because I need an aggregator linked to each subject. this way I can rebuild the original shape and compute a fair test. 

I think it would be nice to have a way for wrapping patients with different shapes into a single dataloader/aggregator. would be cleaner. Anyway, using lists of aggregators also work fine as pointed out by @romainVala, so that this does not look like a critical need.

**Comment by fepegar** [comment]
Thanks for sharing that. Why don't you iterate over your subjects and instantiate a sampler and aggregator for inference at each iteration? In pseudocode, something like:

```
for subject in dataset:
    sampler = GridSampler(...)
    aggregator = GridAggregator(...)
    loader = DataLoader(sampler, ...)
    for batch in loader:
        x, y = prepare_batch(batch)
        yh = model(x)
        aggregator.add_batch(yh)
    subject_loss = compute_loss(subject, yh)
```

Also, why do you use a grid sampler for the Queue? Why not UniformSampler? Are you sure all the patches are being sampled inside the queue? This depends on the images shape and the samples per volume.
